DataGenerator.py

```diff
@@ -0,0 +1,221 @@
+        new_arr = np.zeros(arr.shape)
+        for i in range(len(order)):
+            new_arr[order[i]] = arr[i]
+        return new_arr
+
+    @staticmethod
+    def __generate_data_sample(music_data, speech_data,
+                             n_samples=10000, seq_len=100, embed_dim=128):
+        # Generate random samples by mixing music and speech
+        number_tracks = int(round(seq_len) / 10)
+        data_sample = np.zeros((n_samples, seq_len, embed_dim), dtype=np.float32)
+        mask_sample = np.zeros((n_samples, seq_len, 1), dtype=np.float32)
+        for i in tqdm(range(n_samples)):
+
+            # generate audio features
+            for k in range(number_tracks):
+                rand_idx = np.random.randint(0, len(speech_data))
+                data_sample[i, 10 * k:10 * (k + 1), :] = speech_data[rand_idx, :, :]
+
+            # randomly select position for music part
+            start = np.random.randint(int(-0.2 * seq_len), int(1.2 * seq_len))
+            duration = np.random.randint(seq_len)
+            finish = start + duration
+            if start < 0:
+                start = 0
+            if finish < 0:
+                continue
+            if start < seq_len:
+                if finish > seq_len:
+                    finish = seq_len
+                mask_sample[i, start:finish, 0] = 1.0
+                curr_music_len = finish - start
+                while curr_music_len > 10:
+                    rand_idx = np.random.randint(0, len(music_data))
+                    data_sample[i, start:start + 10, :] = music_data[rand_idx, :, :]
+                    start += 10
+                    curr_music_len = finish - start
+                rand_idx = np.random.randint(0, len(music_data))
+                data_sample[i, start:finish, :] = music_data[rand_idx, :curr_music_len, :]
+        return data_sample, mask_sample
+
+    @classmethod
+    def __extract_music_for_genres(cls, path_to_genres):
+        # prepare music of different genres
+        with open(path_to_genres, "rb") as handle:
+            data_dict_music = pickle.load(handle)
+
+        # data_dict_music has struct {'category_dict', 'embedings_list', 'label_list'}
+        # category_dict - dictionary with genres: blues, classical, etc
+        # embeddings_list - list with embeddings of samples. Every sample has the same len
+        # label_list - lables this sample by category
+        embeddings_list = data_dict_music["embeddings_list"]
+
+        # define some params
+        seq_length = embeddings_list[0].shape[0]
+        embed_dim = embeddings_list[0].shape[1]
+        music_data_new = np.zeros((len(embeddings_list), seq_length, embed_dim), dtype=np.float32)
+        for i in range(len(embeddings_list)):
+            music_data_new[i, :, :] = embeddings_list[i]
+
+        music_data_new = cls.__uint8_to_float32(music_data_new)
+        # reshape 31 frames data to 10 sec frames
+        music_data_new_res = music_data_new.reshape(3100, 10, 128)
```

Comment:
please, do not hardcode values such as embedding size. use variables instead (or constants if applicable)

Reply:
for current realisation the embedding size will be specified in vggish_params.py

Reply:
Done